openhcl/virt_mshv_vtl/src/processor/mshv/tlb_lock.rs

```diff
@@ -50,15 +50,21 @@ impl UhProcessor<'_, HypervisorBacked> {
     /// Check the status of the TLB lock of the target VTL on the current VP.
     pub(crate) fn is_tlb_locked(&mut self, requesting_vtl: Vtl, target_vtl: GuestVtl) -> bool {
         debug_assert_eq!(requesting_vtl, Vtl::Vtl2);
-        let local_status = self.vtls_tlb_locked.get(requesting_vtl, target_vtl);
+        let locally_locked = self.vtls_tlb_locked.get(requesting_vtl, target_vtl);
         // The hypervisor may lock the TLB without us knowing, but the inverse should never happen.
-        if local_status {
-            debug_assert!(self.is_tlb_locked_in_hypervisor(target_vtl))
-        };
-        local_status
+        #[expect(
+            clippy::debug_assert_with_mut_call,
+            reason = "just sanity checking, but the function can update the cached value in the non-debug-assert branch"
+        )]
+        if locally_locked {
+            debug_assert!(self.is_tlb_locked_in_hypervisor(requesting_vtl, target_vtl));
```

Comment:
Can you instead have this call a non-caching version of the function? That way if there's some bug in the cache logic, the assertion won't blow up the state, making this impossible to diagnose.